AdtSampleApp/SampleClientApp/Program.cs

```diff
@@ -3,77 +3,46 @@
-            {
-                int width = Math.Min(Console.LargestWindowWidth, 150);
-                int height = Math.Min(Console.LargestWindowHeight, 40);
-                Console.SetWindowSize(width, height);
-            }
-
+            Uri adtInstanceUrl;
             try
             {
                 // Read configuration data from the 
                 IConfiguration config = new ConfigurationBuilder()
-                    .AddJsonFile("serviceConfig.json", false, true)
+                    .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
                     .Build();
-                clientId = config["clientId"];
-                tenantId = config["tenantId"];
-                adtInstanceUrl = config["instanceUrl"];
+                adtInstanceUrl = new Uri(config["instanceUrl"]);
             }
-            catch (Exception)
+            catch (Exception ex) when (ex is FileNotFoundException || ex is UriFormatException)
             {
-                Log.Error($"Could not read service configuration file serviceConfig.json");
-                Log.Alert($"Please copy serviceConfig.json.TEMPLATE to serviceConfig.json");
-                Log.Alert($"and edit to reflect your service connection settings.");
-                Log.Alert($"Make sure that 'Copy always' or 'Copy if newer' is set for serviceConfig.json in VS file properties");
-                Environment.Exit(0);
+                Log.Error($"Could not read configuration. Have you configured your ADT instance URL in appsettings.json?");
+                throw;
             }
 
             Log.Ok("Authenticating...");
             try
             {
-                var credential = new InteractiveBrowserCredential(tenantId, clientId);
-                client = new DigitalTwinsClient(new Uri(adtInstanceUrl), credential);
-                // force authentication to happen here
-                try
-                {
-                    client.GetDigitalTwin("---");
-                }
-                catch (RequestFailedException)
-                {
-                    // As we are intentionally try to retrieve a twin that is most likely not going to exist, this exception is expected
-                    // We just do this to force the authentication library to authenticate ahead
-                }
-                catch (Exception e)
-                {
-                    Log.Error($"Authentication or client creation error: {e.Message}");
-                    Log.Alert($"Have you checked that the configuration in serviceConfig.json is correct?");
-                    Environment.Exit(0);
-                }
+                var credential = new DefaultAzureCredential();
+                client = new DigitalTwinsClient(adtInstanceUrl, credential);
+
+                // Make a call to the API to force authentication to happen here
+                client.Query("SELECT * FROM DIGITALTWINS");
```

Comment:
I would not run a query here. If the service instance we have already has twins (potentially many) this is a relatively expensive call. I think originally I just had a Get call here. 
More importantly, I actually think we should remove the attempt to force immediate authentication entirely - this is not a current pattern for Azure Identity, as it turns out, and I think I now see that it might be conceptually flawed - even if the auth at this moment in the app worked fine , it does not mean it will work in later calls (for example, after letting the app sit for a long time). I suppose it is probably better to handle auth errors in the remainder of the app.  

Thoughts?

Reply:
The query is paged. You're correct that it may cost more than other calls, but this is the only call where we don't have to make some error handling gymnastics just to prove that we are successfully authenticated and connected. In most cases the ADT instance will however be empty, or nearly empty, since this is a getting started sample.

I do agree though that having a call at all is not desirable. I'm going to remove the call and just have a handler for the AuthException with the helpful link elsewhere.